server/statshouse/worker-stats-buffer.cpp

```diff
@@ -111,6 +111,10 @@ void WorkerStatsBuffer::flush() {
   make_metric(metrics, "kphp_memory_job_common_request_usage", QueryStatKey::job_common_request_memory_usage, tags);
   make_metric(metrics, "kphp_memory_job_common_request_real_usage", QueryStatKey::job_common_request_real_memory_usage, tags);
 
+  if (metrics.empty()) {
+    return;
+  }
```

Comment:
`last_send_time` is not updated here => if metrics remain empty, flush will be called on every subsequent `cron` call. That can be significantly more often than once per second.